Functions/Mock.ps1

```diff
@@ -1544,3 +1547,75 @@ function Remove-MockFunctionsAndAliases {
         & $script:SafeCommands['Remove-Item'] "function:/$($bootstrapFunction.Name)"
     }
 }
+
+
+function Repair-ConflictingParameters {
+    [CmdletBinding()]
+    [OutputType([System.Management.Automation.CommandMetadata])]
+    param(
+        [Parameter(Mandatory)]
+        [System.Management.Automation.CommandMetadata]
+        $Metadata
+    )
+
+    $paramMetadatas = @()
+    $paramMetadatas += $Metadata.Parameters.Values
+
+    foreach ($paramMetadata in $paramMetadatas) {
+        if ($paramMetadata.IsDynamic) {
+            continue
+        }
+
+        $conflictingParams = Get-ConflictingParameterNames
```

Comment:
Normally I would advocate for using functions like you did, but it is easy enough to find where the variable we are sharing was modified, and the functions add a tiny bit of runtime overhead. BUT mostly they make it look more complicated than it is (IMHO). Please replace `Get-ConflictingParameterNames` with just the variable, and the `Get-ParameterPrefixForConflictingParameters` and the script scoped variable, with just using the  underscore in place. The prefix will not change change once a release is done so we don't need the DRYness.

Reply:
Well I introduced the `Get-ConflictingParameterNames` so I can mock it on the tests and use a different array of conflicting values, because if it will fail to run some of the test if *PSEdition* value is used

Reply:
@renehernandez okay, in that case can you get rid of only the prefix function, please?

Reply:
Done